src/components/SignUp.tsx

```diff
@@ -0,0 +1,161 @@
+import { useState } from 'react';
+import { useForm } from 'react-hook-form';
+import {
+  ButtonGroup,
+  FormControl,
+  FormErrorMessage,
+  FormLabel,
+  Input,
+  InputGroup,
+  InputRightElement,
+  Text,
+} from '@chakra-ui/react';
+import { ViewIcon, ViewOffIcon } from '@chakra-ui/icons';
+import AuthButton from './common/Auth/AuthButton';
+import passwordValidation from '../utils/passwordValidation';
+
+export interface SignUpFormValues {
+  email: string;
+  password: string;
+  passwordConfilm: string;
+  nickname: string;
+}
+
+const SignUp = () => {
+  const [isShow, setIsShow] = useState(false);
+  const {
+    handleSubmit,
+    register,
+    formState: { errors },
+    reset,
+  } = useForm<SignUpFormValues>({
+    defaultValues: {
+      email: '',
+      password: '',
+      passwordConfilm: '',
+      nickname: '',
+    },
+  });
+
+  const onSubmit = (data: SignUpFormValues) => {
+    console.log(data);
+  };
+
+  const handlePasswordShow = () => {
+    setIsShow((prev) => !prev);
+  };
+
```

Comment:
```suggestion
useEffect(() => {
    setFocus("email");
  }, [setFocus]);
```

바로 포커스 주는 방식은 어떤가요??